### PR TITLE
[Keyboard] Add DE80 PCB

### PR DESCRIPTION
**Pull request by Deemen17**
## Description

Add QMK support for DE80 PCB 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/deemen17/de80/keyboard.json [hunk 1]
```diff
@@ -0,0 +1,997 @@
...
+            "twinkle": true
+        },
+        "led_count": 40,
+        "max_brightness": 250,
+        "sleep": true
+    },
+    "ws2812": {
+        "driver": "spi",
+        "pin": "B15"
+    },
```
Comment (dunk2k) [inline keyboards/deemen17/de80/keyboard.json:49]:
```suggestion
    },
    "layout_aliases": {
        "LAYOUT_all": "LAYOUT_tkl_f13_ansi_tsangan_split_bs_rshift"
    },
```

File: keyboards/deemen17/de80/keyboard.json [hunk 2]
```diff
@@ -0,0 +1,997 @@
...
+                {"matrix": [5, 6], "x": 4, "y": 5.25, "w": 7},
+                {"matrix": [5, 11], "x": 11, "y": 5.25, "w": 1.5},
+                {"matrix": [5, 12], "x": 12.5, "y": 5.25},
+                {"matrix": [5, 13], "x": 13.5, "y": 5.25, "w": 1.5},
+
+                {"matrix": [5, 14], "x": 15.25, "y": 5.25},
+                {"matrix": [5, 15], "x": 16.25, "y": 5.25},
+                {"matrix": [5, 16], "x": 17.25, "y": 5.25}
+            ]
+        },
```
Comment (dunk2k) [inline keyboards/deemen17/de80/keyboard.json:163]: replaced with `layout_aliases`
```suggestion
```

File: keyboards/deemen17/de80/keyboard.json [hunk 3]
```diff
@@ -0,0 +1,997 @@
...
+    "ws2812": {
+        "driver": "spi",
+        "pin": "B15"
+    },
+    "community_layouts": [
+        "tkl_ansi_tsangan", 
+        "tkl_ansi_tsangan_split_bs_rshift",
+        "tkl_f13_ansi_tsangan",
+        "tkl_f13_ansi_tsangan_split_bs_rshift"
+    ],
```
Comment (dunk2k) [inline keyboards/deemen17/de80/keyboard.json:55]: add compatible Community Layouts
```suggestion
    "community_layouts": ["tkl_ansi_tsangan", "tkl_ansi_tsangan_split_bs_rshift", "tkl_ansi_wkl", "tkl_ansi_wkl_split_bs_rshift", "tkl_f13_ansi_tsangan", "tkl_f13_ansi_tsangan_split_bs_rshift", "tkl_f13_ansi_wkl", "tkl_f13_ansi_wkl_split_bs_rshift" ],
```

**Review by dunk2k — CHANGES_REQUESTED** [review]
layout changes

File: keyboards/deemen17/de80/keyboard.json [hunk 4]
```diff
@@ -0,0 +1,888 @@
...
+        "mousekey": true,
+        "nkro": true,
+        "rgblight": true
+    },
+    "matrix_pins": {
+        "rows": ["A8", "A10", "A9", "B11", "C14", "C13"],
+        "cols": ["B14", "B13", "B12", "B2", "B1", "B0", "A7", "A6", "A5", "A4", "A3", "A2", "A1", "A0", "A15", "B7", "B6"]
+    },
+    "processor": "STM32F103",
+    "board": "STM32_F103_STM32DUINO",
```
Comment (zvecr) [inline keyboards/deemen17/de80/keyboard.json:19]:
```suggestion
```

File: keyboards/deemen17/de80/readme.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+    make deemen17/de80:default:flash
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
+
+## How to flash new firmware
+
+**Step 1:** Get the.uf2 file
+
+**Step 2:** Enter the bootloader in 3 ways
```
Comment (zvecr) [inline keyboards/deemen17/de80/readme.md:25]:
```suggestion
## Bootloader

Enter the bootloader in 3 ways:
```

File: keyboards/deemen17/de80/readme.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+| Column | 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12 | 13 | 14 | 15 | 16 |
+| -------- | ------- | ------- | ------- | ------- | ------- | ------- | ------- | ------- | ------- | ------- | ------- | ------- | ------- | ------- | ------- | ------- | ------- |
+| Pin | B14 | B13 | B12 | B2 | B1 | B0 | A7 | A6 | A5 | A4 | A3 | A2 | A1 | A0 | A15 | B7 | B6 |
+
+### Row Pin Configuration
+
+| Row | 0 | 1 | 2 | 3 | 4 | 5 |
+| ------- | ------- | ------- | ------- | ------- | ------- | ------- |
+| Pin | A8 | A10 | A9 | B11 | C14 | C13 |
```
Comment (zvecr) [inline keyboards/deemen17/de80/readme.md:58]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Comment by Deemen17** [comment]
@zvecr thanks for your suggestion.

**Review by drashna — APPROVED** [review]

File: keyboards/deemen17/de80/readme.md [hunk 7]
```diff
@@ -0,0 +1,27 @@
+# DE80
+
+![deemen17/de80](https://i.imgur.com/oZvFM0G.png)
```
Comment (zvecr) [inline keyboards/deemen17/de80/readme.md:3]: This image shows an encoder, but this PR has no config for one?

Comment (Deemen17) [reply keyboards/deemen17/de80/readme.md:3]: Thanks for catching that. I'll add that in now.

**Review by fauxpark — APPROVED** [review]

**Comment by Deemen17** [comment]
Thank all of you for approve and merge DE80! @dunk2k @zvecr @drashna @fauxpark

**Pull request merged**